sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -49,11 +52,19 @@ impl BlobClient {
             .per_try_policies
             .push(Arc::new(oauth_token_policy) as Arc<dyn Policy>);
 
-        let client = GeneratedBlobClient::new(endpoint, credential, Some(options))?;
+        let client = GeneratedBlobClient::new(
+            endpoint,
+            credential.clone(),
+            "2025-05-05".to_string(),
+            container_name.clone(),
+            blob_name.clone(),
+            Some(options),
+        )?;
```

Comment:
Now that we that we don't have an accessor `BlobClient` and this literally the generated client that will perform the REST operations, I think this makes some fields redundant on our handwritten clients.

Generated `BlobClient` struct:
```rust
pub struct BlobClient {
    pub(crate) blob: String,
    pub(crate) container_name: String,
    pub(crate) endpoint: Url,
    pub(crate) pipeline: Pipeline,
    pub(crate) version: String,
}
```
We can now omit `blob_name`, `container_name`, `endpoint` from the handwritten client layer and instead do `self.client.<field>`

Reply:
Omit from struct, add getters (that will go 1 layer down into gen client)